### DISCUSSION TITLE
Airflow Web UI Lag and Unresponsive Webserver on EC2 Instance

### DISCUSSION THREAD

**Discussion by AditModi** [apache/airflow#28986] [Q&A] [answered]
Hello, 

I have recently set up an Airflow instance on an EC2 instance using Terraform and user data. The installation was successful, however, when I try to create DAGs, my web UI becomes slow and I am unable to restart the Airflow webserver. The scheduler seems to be working fine, but I am unable to access the web UI and my DAGs. Can anyone suggest some options to work around this issue?

**Comment by potiuk** [comment]
Increase memory for your machine maybe ? Generally monitor your resource usage and see if they cause any problems (like with any other app).

**Comment by AditModi** [comment]
I don't think memory is the cause of the problem. checked the instance and it seems that the instance itself is working just fine.

**Reply by potiuk** [reply]
Then maybe other resources. I was just guessing. Networking, DB or anything else might be a problem.

It's impossible to guess it right without you looking at it  :).

**Comment by AditModi** [comment, marked as answer]
okay, let me take a much deeper dive at other resources,

**Reply by AditModi** [reply]
Looking closely, there were a couple of reasons for it not working as expected. (the user data scripts were not working as intended and it caused the lag issue) - after fixing the scripts, I can confirm that everything is working as expected.
